jenkins-scripts/tools/outdated-job-runner.groovy

```diff
@@ -0,0 +1,223 @@
+    'ignition_gui-ci-ign-gui6-homebrew-amd64',
+    'ignition_tools-ci-ign-tools1-homebrew-amd64',
+    'ignition_gazebo-ci-ign-gazebo3-focal-amd64',
+    'ignition_cmake-ci-ign-cmake2-focal-amd64',
+    'ign_sensors-gz-7-win',
+    'ignition_fuel-tools-ci-gz-fuel-tools8-focal-amd64',
+    'ignition_rendering-ci-ign-rendering6-homebrew-amd64',
+    'ign_msgs-ign-8-win',
+    'ignition_msgs-ci-gz-msgs9-focal-amd64',
+    'ignition_gazebo-ci-ign-gazebo6-homebrew-amd64',
+    'ign_gui-gz-7-win',
+    'sdformat-sdf-9-win',
+    'ignition_transport-ci-gz-transport12-homebrew-amd64',
+    'ignition_gazebo-ci-ign-gazebo6-focal-amd64',
+    'ignition_launch-ci-gz-launch6-focal-amd64',
+    'ignition_fuel-tools-ci-ign-fuel-tools4-focal-amd64',
+    'ign_cmake-gz-3-win',
+    'ign_gazebo-ign-6-win',
+    'ignition_sensors-ci-ign-sensors3-focal-amd64',
+    'sdformat-ci-sdformat13-homebrew-amd64',
+    'sdformat-ci-sdformat13-focal-amd64',
+    'ignition_common-ci-gz-common5-homebrew-amd64',
+    'ignition_cmake-ci-gz-cmake3-focal-amd64',
+    'ign_transport-ign-11-win',
+    'ign_gazebo-gz-7-win',
+    'ign_sensors-ign-6-win',
+    'ignition_sensors-ci-ign-sensors6-homebrew-amd64',
+    'ignition_gazebo-ci-gz-sim7-homebrew-amd64',
+    'ignition_gui-ci-gz-gui7-homebrew-amd64',
+    'ignition_math-ci-gz-math7-focal-amd64',
+    'ignition_transport-ci-ign-transport11-homebrew-amd64',
+    'ignition_math-ci-gz-math7-homebrew-amd64',
+    'ignition_fuel-tools-ci-ign-fuel-tools7-homebrew-amd64'
+];
+
+def trackedJobs = new LinkedHashSet(trackedJobsList);
+
+def runJob(String jobName) {
+    /*
+     * Schedule a job to run immediately
+     */
+    def job = Jenkins.instance.getItemByFullName(jobName)
+    job.scheduleBuild(0)
+}
+
+def findAvailableNodes() {
+    /*
+     * Find available nodes to run jobs per OS using labels
+     * win: windows
+     * osx: macos
+     * linux: docker
+     */
+    def availableNodes = [osx: [], win: [], linux: []]
+
+    Jenkins.instance.nodes.each { node ->
+        if (node.computer.online && node.computer.countIdle() > 0) {
+            if (node.getLabelString().contains("win")) {
+                availableNodes.win << node.name
+            } else if (node.getLabelString().contains("osx")) {
+                availableNodes.osx << node.name
+            } else if (node.getLabelString().contains("docker")) {
+                availableNodes.linux << node.name
+            }
+        }
+    }
+
+    return availableNodes
+}
+
+def runJobsInAvailableNodes(LinkedHashMap outdatedJobs) {
+    /*
+     * For each node in each OS, run a job if there are jobs to run
+     */
+    def availableNodes = findAvailableNodes()
+
+    availableNodes.each { os -> 
+        os.value.each { node ->
+                if (outdatedJobs[os.key].size() > 0) {
+                    def jobName = outdatedJobs[os.key].remove(0)
+                    println("Running job ${jobName} in node ${node}")
+                    runJob(jobName)
```

Comment:
Can we please add an error checking mechanism in the case that there is a job name that does not exists in the list or other problem arise when calling the `scheduleBuild`. We need to think about failing the build or making it unstable in that scenarios. For making it unstable, please grep for `MARK_BUILD_UNSTABLE` in the DSL code to see how that can be done.

Reply:
I see https://github.com/gazebo-tooling/release-tools/blob/master/jenkins-scripts/tools/label-assignment-backstop.groovy#L37, so I added `println("MARK_AS_UNSTABLE")` instead.